.gitlab-ci.yml

```diff
@@ -9,118 +9,136 @@ include:
-#    - docker logs -f server-mysql
-#    - echo "Ignoring known MySQL mismatch 'ChannelMembers.SchemeGuest'"
-#    - docker-compose --no-ansi exec -T mysql mysql -D migrated -uroot -pmostest -e "ALTER TABLE ChannelMembers DROP COLUMN SchemeGuest;" || echo "drop failed"
-#    - docker-compose --no-ansi exec -T mysql mysql -D latest -uroot -pmostest -e "ALTER TABLE ChannelMembers DROP COLUMN SchemeGuest;" || echo "drop failed"
-#    - echo "Generating dump"
-#    - docker-compose --no-ansi exec -T mysql mysqldump --skip-opt --no-data --compact -u root -pmostest migrated > migrated.sql
-#    - docker-compose --no-ansi exec -T mysql mysqldump --skip-opt --no-data --compact -u root -pmostest latest > latest.sql
-#    - echo "Removing databases created for db comparison"
-#    - docker-compose --no-ansi exec -T mysql mysql -uroot -pmostest -e 'DROP DATABASE migrated; DROP DATABASE latest'
-#    - echo "Generating diff"
-#  rules:
-#    - if: '$TEST == "yes"'
+lint:
+  image: $CI_REGISTRY/mattermost/ci/images/golangci-lint:v1.39.0-1
+  stage: test
+  script:
+    - GO111MODULE=off GOBIN=$PWD/bin go get -u github.com/mattermost/mattermost-govet
+    - make config-reset
+    - make check-style #todo MM_VET_OPENSPEC_PATH='$CI_PROJECT_DIR/mattermost-api-reference/v4/html/static/mattermost-openapi-v4.yaml'
+  timeout: 60 mins
+  rules:
+    - if: '$CI_COMMIT_REF_NAME == "master"'
+
+layers:
+  image: $CI_REGISTRY/mattermost/ci/images/mattermost-build-server:20201119_golang-1.15.5
+  stage: test
+  script:
+    - make store-layers
+    - if [[ -n $(git status --porcelain) ]]; then echo "Please update the store layers using make store-layers"; exit 1; fi
+    - git reset --hard
+    - make app-layers
+    - if [[ -n $(git status --porcelain) ]]; then echo "Please update the app layers using make app-layers"; exit 1; fi
+    - git reset --hard
+  rules:
+    - if: '$CI_COMMIT_REF_NAME == "master"'
+
+.test-mysql: # commented due to high memory consumption for runners
+  stage: test
+  image: $CI_REGISTRY/mattermost/ci/images/mattermost-build-docker:19.03.14-1
+  services:
+    - name: $CI_REGISTRY/mattermost/ci/images/docker-dind:19.03.14-1
+      alias: docker
+  variables:
+    DOCKER_TLS_CERTDIR: ""
+    DOCKER_HOST: tcp://docker:2375
+    DOCKER_DRIVER: overlay2
+    DOCKER_CONTENT_TRUST: 0
+    DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE: ""
+  before_script:
+    - docker login ${CI_REGISTRY} --username ${CI_REGISTRY_USER} --password ${CI_REGISTRY_PASSWORD}
+  script:
+    - export COMPOSE_PROJECT_NAME="${CI_PIPELINE_IID}mysql"
+    - export RACE_MODE=""
+    - time cat .gitlab-ci/scripts/test/mysql.sh | /bin/bash
+  artifacts:
+    when: always
+    reports:
+      junit: report.xml
+    paths:
+      - build/logs
+  timeout: 2 hours
+  tags:
+    - k8s-memory
+  rules:
+    - if: '$CI_COMMIT_REF_NAME == "master"'
+
+.test-postgres: # commented due to high memory consumption for runners
```

Comment:
I remember that the tests were failing due to some resource issue before. It seemed to be a network issue but that wasn't the case. (i.e `nc: bad address` issue) Is this the same problem? Also, when I check the `scripts/test/postgres.sh` did running the tests under the same network with the composed services solved the issue?

Reply:
It was actually a docker network issue when multiple pipelines were running. 
These are the bits which solved it:
https://github.com/mattermost/mattermost-server/blob/77e55f562a03d1bcf28505f78823ab418cc2f939/.gitlab-ci.yml#L117
https://github.com/mattermost/mattermost-server/blob/77e55f562a03d1bcf28505f78823ab418cc2f939/.gitlab-ci/scripts/test-schema/mysql.sh#L6
https://github.com/mattermost/mattermost-server/blob/testing-in-gitlab/.gitlab-ci/scripts/test-schema/mysql.sh#L9
https://github.com/mattermost/mattermost-server/blob/77e55f562a03d1bcf28505f78823ab418cc2f939/build/gitlab-dc.schemamysql.yml#L36

And for talking to each other when the containers are on the same network:
https://github.com/mattermost/mattermost-server/blob/77e55f562a03d1bcf28505f78823ab418cc2f939/build/gitlab-dc.schemamysql.yml#L22

Reply:
Does pointing out the code bits involved to fix the issues with the docker network explain it? I can also try to explain it better. Basically the docker-compose.common.yml has one docker network right now called `mm-test`. This works for CircleCI, since we are not trying to run multiple docker-compose projects on the same machine. Every time a docker-compose is run, a new machine is created by CircleCI. As soon as we are trying to run multiple docker-compose projects on the same machine (not really, but for explaining it, yes), we need to provide unique identifiers and start all the containers which need to talk to each other in the appropriately uniquely identified network and alias the container so the service can be talked to. Is this making it clearer?

Reply:
Thanks for the explanation @metanerd. It seemed to me a different error when I logged the errors and that's why I asked. The container was not starting at all due to some superficial read/write access error. But if it is working, it's fine for me 😁 On the other hand, just out of curiosity; how did the `mm-test` network persisted all the time? I mean if a pipeline finishes, shouldn't it get wiped and removed?

Reply:
The problem was that the `mm-test` network was overloaded with multiple containers all trying to expose same ports, if I understood it correctly :smile: . Now one environment is isolated to one docker network. And yeah, it works.

Reply:
Awesome 🎉